### PR TITLE
Fix todo overlay

### PR DESCRIPTION
**Pull request by NopoTheGamer**
doesnt fix god pot yet because 1 that regex looks like aids 2 i am going to sleep

in theory fixes the cookie notif but not 100% sure

### CODE REVIEW

**Comment by heyngra** [comment]
https://github.com/heyngra/NotEnoughUpdates/commit/5f44cf38284fb5bd426dc02beb7c23a165656e78 
can't commit it here, but here's the god pot fix (it should works, works fine in my dev env)

**Comment by hannibal002** [comment]
Is the god pot time in the API?

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java [hunk 1]
```diff
@@ -336,50 +342,48 @@ public void update() {
...
+									hidden.cookieBuffRemaining += val * 60 * 1000;
+									break;
+								case "Seconds":
+								case "Second":
+								case "s":
+									hidden.cookieBuffRemaining += val * 1000;
+									break;
 							}
+						} catch (NumberFormatException e) {
+							hidden.cookieBuffRemaining = 0;
```
Comment (Lulonaut) [inline src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java:385]: Woudn't it be helpful to tell the user that something went wrong or at least log it?

**Review by Lulonaut — DISMISSED** [review]

**Review by Lulonaut — DISMISSED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/CookieWarning.java [hunk 2]
```diff
@@ -74,22 +74,31 @@ public static void checkCookie() {
...
+								break;
+							case "Days":
+								minutes += val * 1440;
+								break;
+							case "Hours":
+								minutes += val * 60;
+								break;
+							case "Minutes":
+								minutes += val;
+								break;
```
Comment (heyngra) [inline src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/CookieWarning.java:101]: shouldn't this also include changes from https://github.com/NotEnoughUpdates/NotEnoughUpdates/pull/432/commits/99a0fe53a9d2f0973dd210d5f022642c992d9ad5 https://github.com/NotEnoughUpdates/NotEnoughUpdates/pull/433/commits/877c7f57f8cf4030fc25d996aad9a17e7d564cf6?

Comment (NopoTheGamer) [reply src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/CookieWarning.java:101]: perhaps

**Review by heyngra — DISMISSED** [review]

**Review by Lulonaut — DISMISSED** [review]
@[NopoTheGamer](https://github.com/NopoTheGamer) NopoTheGamer dismissed [Lulonaut](https://github.com/Lulonaut)’s [stale review](https://github.com/NotEnoughUpdates/NotEnoughUpdates/pull/432#pullrequestreview-1177355322)

**Review by Ascynx — APPROVED** [review]
we do a small amount of trolling

**Review by heyngra — DISMISSED** [review]

**Comment by hannibal002** [comment]
```
java.lang.NumberFormatException: null
	at java.lang.Integer.parseInt(Integer.java:542)
	at java.lang.Integer.parseInt(Integer.java:615)
	at io.github.moulberry.notenoughupdates.overlays.TimersOverlay.update(TimersOverlay.java:316)
	at io.github.moulberry.notenoughupdates.overlays.TextTabOverlay.tick(TextTabOverlay.java:45)
	at io.github.moulberry.notenoughupdates.listener.NEUEventListener.onTick(NEUEventListener.java:184)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_8_NEUEventListener_onTick_ClientTickEvent.invoke(.dynamic)
```

**Comment by hannibal002** [comment]
grafik
grafik

Does not work for me

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java [hunk 3]
```diff
@@ -307,79 +309,95 @@ public void update() {
...
-						}
-						if (godpotRemaingTimeUnformatted.length >= 2) {
-							godPotDuration = godPotDuration + (long) Integer.parseInt(godpotRemaingTimeUnformatted[i]) * 60 * 1000;
-							i++;
-						}
-						if (godpotRemaingTimeUnformatted.length >= 1) {
-							godPotDuration = godPotDuration + (long) Integer.parseInt(godpotRemaingTimeUnformatted[i]) * 1000;
+						long godpotRemainingTime;
+							for (int i = 1; i < activeEffectsMatcher.groupCount(); i += 2) {
+								godpotRemainingTime = Integer.parseInt(activeEffectsMatcher.group(i));
```
Comment (hannibal002) [inline src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java:316]: NPE

Comment (heyngra) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java:316]: I have no idea why, but suddenly the regex returns two more groups that contain only nulls
image
image

Comment (hannibal002) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java]: Looks like Hypixel changed the format again?

Comment (heyngra) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java]: https://github.com/NotEnoughUpdates/NotEnoughUpdates/pull/441

**Comment by jani270** [comment]
Bro did you really merge wrong

**Comment by hannibal002** [comment]
Now it looks like hell <3

**Comment by jani270** [comment]
bro please just stop
why

**Comment by jani270** [comment]
So useless

**Comment by jani270** [comment]
This pullrequest is chatting

**Review by hannibal002 — DISMISSED** [review]
Does work for me

**Review by romangraef — DISMISSED** [review]
Still a downgrade from the old system, but the best we can do
image

**Review by romangraef — APPROVED** [review]

**Review by Lulonaut — APPROVED** [review]

**Comment by heyngra** [comment]
legend says that someday this will be merged

**Comment by romangraef** [comment]
@heyngra can you request reviews or can only members do that?
image

**Review by romangraef — APPROVED** [review]

**Comment by heyngra** [comment]
> @heyngra can you request reviews or can only members do that? image

no, neither I am a member, neither this is my pull request, and I can't
image

**Comment by romangraef** [comment]
oh you arent the PR author. that is probably why

**Comment by Ascynx** [comment]
> > > oh you arent the PR author. that is probably why
> > 
> > 
> > Nope, had tried on my old prs and it's not a permission non-members have
> 
> image I mean I requested a review before, so I guess it's a permission that non-members have, if PR is opened and it's theirs

that's using re-request of a review and not an actual review request.

**Review by hannibal002 — APPROVED** [review]

**Comment by hannibal002** [comment]
Yes please, contributorzz!

image

**Pull request merged**